ml-agents/mlagents/trainers/policy/tf_policy.py

```diff
@@ -69,20 +69,26 @@ def __init__(
         self.update_dict: Dict[str, tf.Tensor] = {}
         self.sequence_length = 1
         self.seed = seed
-        self.brain = brain
+        self.behavior_spec = behavior_spec
 
-        self.act_size = brain.vector_action_space_size
-        self.vec_obs_size = brain.vector_observation_space_size
-        self.vis_obs_size = brain.number_visual_observations
+        self.act_size = (
+            list(behavior_spec.discrete_action_branches)
+            if behavior_spec.is_action_discrete()
+            else [behavior_spec.action_size]
+        )
+        self.vec_obs_size = np.sum(
```

Comment:
nit: I think `sum(shape[0] for shape in behavior_spec.observation_shapes if len(shape) == 1)` is a bit better since it doesn't require creating the intermediate list. Same for vis_obs_size below.